openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -43,6 +43,25 @@
 invalid_not_found_fmt = "XBlock {usage_key} does not exist, or you don't have permission to view it."
 
 
+def parse_version_request(version_str: str | None) -> LatestVersion | int:
+    """
+    Given a version parameter from a query string (?version=14, ?version=draft,
+    ?version=published), get the LatestVersion parameter to use with the API.
+    """
+    if version_str is None:
+        return LatestVersion.AUTO  # AUTO = published if we're in the LMS, draft if we're in Studio.
+    if version_str == "draft":
+        return LatestVersion.DRAFT
+    if version_str == "published":
+        return LatestVersion.PUBLISHED
+    try:
+        return int(version_str)
+    except ValueError:
+        raise serializers.ValidationError(  # pylint: disable=raise-missing-from
+            "Invalid version specifier '{version_str}'. Expected 'draft', 'published', or an integer."
+        )
```

Comment:
nitpick: not a fan of hungarian notation. not sure if there's value in using it here.
```suggestion
def parse_version_request(version: str | None) -> LatestVersion | int:
    """
    Given a version parameter from a query string (?version=14, ?version=draft,
    ?version=published), get the LatestVersion parameter to use with the API.
    """
    if version is None:
        return LatestVersion.AUTO  # AUTO = published if we're in the LMS, draft if we're in Studio.
    if version == "draft":
        return LatestVersion.DRAFT
    if version == "published":
        return LatestVersion.PUBLISHED
    try:
        return int(version)
    except ValueError:
        raise serializers.ValidationError(  # pylint: disable=raise-missing-from
            "Invalid version specifier '{version}'. Expected 'draft', 'published', or an integer."
        )
```

Reply:
In some parts of the code we have `version` with type `LatestVersion | int` so I thought it's confusing if other parts of the code have `version` with type `str | None` which is quite different. I could call it `version_request` if that satisfies your aversion to Hungarian notation.

Reply:
ohh that's a good reason, so I have no issues with keeping it